src/uu/df/df.rs

```diff
@@ -0,0 +1,878 @@
+#![crate_name = "uu_df"]
+
+/*
+ * This file is part of the uutils coreutils package.
+ *
+ * (c) Fangxu Hu <framlog@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE file
+ * that was distributed with this source code.
+ */
+
+extern crate clap;
+extern crate libc;
+#[macro_use]
+extern crate uucore;
+
+#[cfg(windows)]
+extern crate kernel32;
+#[cfg(windows)]
+extern crate winapi;
+
+use clap::{App, Arg};
+#[cfg(windows)]
+use kernel32::{
+    FindFirstVolumeW, FindNextVolumeW, FindVolumeClose, GetDiskFreeSpaceW, GetDriveTypeW,
+    GetLastError, GetVolumeInformationW, GetVolumePathNamesForVolumeNameW, QueryDosDeviceW,
+};
+#[cfg(unix)]
+#[cfg(target_os = "macos")]
+use libc::statfs;
+use std::cell::Cell;
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::ffi::CString;
+#[cfg(unix)]
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+#[cfg(windows)]
+use std::os::windows::prelude::*;
+use std::{env, mem};
+
+static VERSION: &str = env!("CARGO_PKG_VERSION");
+static ABOUT: &str = "Show information about the file system on which each FILE resides,\n\
+                      or all file systems by default.";
+
+const EXIT_OK: i32 = 0;
+
+#[cfg(windows)]
+const MAX_PATH: usize = 266;
+
+#[cfg(target_os = "linux")]
+static LINUX_MOUNTINFO: &str = "/proc/self/mountinfo";
+#[cfg(target_os = "linux")]
+static LINUX_MTAB: &str = "/etc/mtab";
+
+static OPT_ALL: &str = "all";
+static OPT_BLOCKSIZE: &str = "blocksize";
+static OPT_DIRECT: &str = "direct";
+static OPT_TOTAL: &str = "total";
+static OPT_HUMAN_READABLE: &str = "human-readable";
+static OPT_HUMAN_READABLE_2: &str = "human-readable-2";
+static OPT_INODES: &str = "inodes";
+static OPT_KILO: &str = "kilo";
+static OPT_LOCAL: &str = "local";
+static OPT_NO_SYNC: &str = "no-sync";
+static OPT_OUTPUT: &str = "output";
+static OPT_PORTABILITY: &str = "portability";
+static OPT_SYNC: &str = "sync";
+static OPT_TYPE: &str = "type";
+static OPT_PRINT_TYPE: &str = "print-type";
+static OPT_EXCLUDE_TYPE: &str = "exclude-type";
+static OPT_VERSION: &str = "version";
+
+static MOUNT_OPT_BIND: &str = "bind";
+
+/// Store names of file systems as a selector.
+/// Note: `exclude` takes priority over `include`.
+struct FsSelector {
+    include: HashSet<String>,
+    exclude: HashSet<String>,
+}
+
+struct Options {
+    show_local_fs: bool,
+    show_all_fs: bool,
+    show_listed_fs: bool,
+    show_fs_type: bool,
+    show_inode_instead: bool,
+    print_grand_total: bool,
+    block_size: usize,
```

Comment:
```suggestion
    // block_size: usize,
```
`cargo clippy` says 'block_size' isn't used; changed to comment if you think it might be used later.